bucket.go

```diff
@@ -473,18 +486,18 @@ type UpdateFunc func(current []byte) (updated []byte, expiry *uint32, delete boo
 
 // UpdatedDoc is returned by WriteUpdateWithXattrsFunc, to indicate the new document value and xattrs.
 type UpdatedDoc struct {
-	Doc         []byte               // Raw value of the document
-	Xattrs      map[string][]byte    // Each xattr found with its value. If the xattr is not specified by Update or not present in the document, it will be missing from the document
-	IsTombstone bool                 // IsTombstone is true if the document is a tombstone
-	Expiry      *uint32              // Expiry is non-nil to set an expiry
-	Spec        []MacroExpansionSpec // Spec represents which macros to expand
+	Doc            []byte               // Raw value of the document
+	Xattrs         map[string][]byte    // Each xattr found with its value. If the xattr is specified, it will be preserved.
+	XattrsToDelete []string             // xattrs to delete. This is only valid for non tombstone docs.
```

Comment:
Does "non-tombstone" docs refer to:
- documents that are already CBS tombstones in the bucket, or
- documents that are being tombstoned in the current mutation
(or both)

Reply:
This means documents that are going to become tombstones.